### DISCUSSION TITLE
Support for RS485 to ETH devices

### DISCUSSION THREAD

**Discussion by arieltools** [tagyoureit/nodejs-poolController#664] [Q&A] [answered]
Hi there!

Would someone have any insight on how to connect an IntelliFlo using a device like this one https://www.waveshare.com/wiki/RS485_TO_ETH_(B) - RS485 to Ethernet device.

Didn't find anything while searching on the issues, discussions or the [wishlist](https://github.com/users/tagyoureit/projects/1/views/1)

Threads / stuff I've checked:
- https://github.com/tagyoureit/nodejs-poolController/discussions/507
- https://github.com/tagyoureit/nodejs-poolController/discussions/441
- https://www.troublefreepool.com/threads/pentair-intellicenter-pool-control-dashboard-instructional-guide.218514/page-4#post-1927837
- https://github.com/cminyard/ser2net
- https://github.com/tagyoureit/nodejs-poolController/wiki/RS-485-Adapter-Details (goes in-depth into how to test and troubleshoot RS485 connections which helped me diagnose the ETH-to-RS485 adapter)

Awesome Project, thanks!

**Comment by rstrouse** [comment, marked as answer]
Simply set it up as a RS485 Serial TCP Server then set up the connection in the dashPanel.  Set the IP address to the x's and whatever port the adapter is on and you are off to the races.

![image](https://user-images.githubusercontent.com/47839015/208717100-94583717-a7b3-4380-9415-dff4d7296241.png)

**Comment by arieltools** [comment]
Thanks, 
I do get "info: Net connect (socat) 0 connected to: XX.XX.XX.XX:4196"

However, I'm getting a bunch of errors/warns like these:
warn: Message aborted after 3 attempt(s): 165,0,96,33,7,0,1,45
error: Error sending requestPumpStatus for Pump 1: Message aborted after 3 attempt(s): 165,0,96,33,7,0,1,45
warn: Message aborted after 2 attempt(s): 165,0,96,33,4,1,255,2,42
error: Error sending setPumpToRemoteControl for Pump 1: Message aborted after 2 attempt(s): 165,0,96,33,4,1,255,2,42
warn: Message aborted after 2 attempt(s): 165,0,96,33,6,1,10,1,55
error: Error sending setDriveState for Pump 1 : Message aborted after 2 attempt(s): 165,0,96,33,6,1,10,1,55


any clues?

**Reply by rstrouse** [reply]
Tell me about your system.  Do you have an RS485 pump connected?  Is there an EasyTouch, IntelliTouch, or IntelliCenter panel on the bus?

**Comment by arieltools** [comment]
Got it working. The cable got disconnected for some reason so there's that. Thanks for the help.

Pretty straightforward.

**Reply by rstrouse** [reply]
Good to hear.  Merry Christmas!
